lib/load-simple-icons.js

```diff
@@ -26,14 +27,16 @@ function loadSimpleIcons() {
     // (e.g. 'Hive'). If a by-title reference we generate for
     // backwards compatibility collides with a proper slug from Simple Icons
     // then do nothing, so that the proper slug will always map to the correct icon.
-    if (!(title in originalSimpleIcons)) {
-      simpleIcons[title] = icon
+    // Starting in v7, the exported object with the full icon set has updated the keys
+    // to include a lowercase `si` prefix, and utilizes proper case naming conventions.
+    if (!(`si${title}` in originalSimpleIcons)) {
+      simpleIcons[title.toLowerCase()] = icon
     }
-    if (!(legacyTitle in originalSimpleIcons)) {
-      simpleIcons[legacyTitle] = icon
+    if (!(`si${legacyTitle}` in originalSimpleIcons)) {
+      simpleIcons[legacyTitle.toLowerCase()] = icon
     }
 
-    simpleIcons[key] = icon
+    simpleIcons[key.slice(2).toLowerCase()] = icon
```

Comment:
I was just having a look at this myself, but you got in ahead of me :D

Instead of thinking about this in terms of slicing and dicing the object key, should we move to using the slug property?

i.e: one way of thinking about this is

```js
Object.keys(originalSimpleIcons).forEach(key => {
  // handle special cases
  simpleIcons[key.slice(2).toLowerCase()] = icon
})
```

but maybe a better approach would be to think of it as

```js
Object.values(originalSimpleIcons).forEach(icon => {
  const slug = icon.slug
  // handle special cases
  simpleIcons[slug] = icon
})
```

?

Reply:
No opposition to that if you want to give it a whirl, but part of why we had to do this starting in v6 was to maintain backwards compatibility with the slugs. My suspicion is that dropping support for the key would essentially break/eliminate what we put here to avoid that.

Reply:
In SI 6 and below (back to the point they introduced the slugs), the object key was the slug

In SI 7 I'm not clear that `key.slice(2).toLowerCase()` is necessarily the same thing as the slug in all cases. I think we need to dig into the SI source to answer this one.

The things we want to ensure work are:

- The SI slug: as that is what they expose/encourage their users to reference in the lib, via CDN and it is what they use for the filenames on https://simpleicons.org/
- Shields.io's old slugs that we made up before SI introduced a slug property (as far as possible)

Reply:
Maybe I misunderstood what you meant earlier by the "handle special cases" inline comment in your example. Are you simply suggesting that for the object our module exports, that the keys we set are based on the `slug` field of the icon objects instead of the top-level key names on the SI exported object?

Reply:
Maybe it would be clearer if I wrote it like this:

```js
Object.keys(originalSimpleIcons).forEach(key => {

  // some more code in here - chopped out for the purposes of making the example smaller

  simpleIcons[key.slice(2).toLowerCase()] = icon
})
```


```js
Object.values(originalSimpleIcons).forEach(icon => {
  const slug = icon.slug

  // some more code in here - chopped out for the purposes of making the example smaller

  simpleIcons[slug] = icon
})
```

Basically,  instead of assuming that the slug is `key.slice(2).toLowerCase()` (which might be true - but I've not checked), I think we should use the `slug` property. Partly just for clarity of the code and partly because it feels like more robust assumption.

Reply:
> Basically, instead of assuming that the slug is key.slice(2).toLowerCase() (which might be true - but I've not checked), I think we should use the slug property. Partly just for clarity of the code and partly because it feels like more robust assumption.

Gotcha, that wasn't what I'd initially interpreted your previous comment as meaning but I'm with you now. I think this is fair and probably the right way to go, although it does bring up another potential question.

I was looking at this from the perspective of compatibility with v6 keys vs. slugs specifically, so I think for me the questions are whether there were any v6 keys that don't have a matching `slug` in v7, as well as your question as to whether there are any keys in v7 that don't "match" to their slug. I think these can both be fairly easily checked with some quick scripts to compare the two sets so I'll post an update with findings.

Barring any unexpected, weird findings, I'm good moving forward with your proposal to use the slugs

Reply:
> I think these can both be fairly easily checked with some quick scripts to compare the two sets

Indeed. I decided to check it. I ran


## SimpleIcons 6:

```js
import simpleIcons from 'simple-icons'

Object.keys(simpleIcons).forEach(key => {
  if (key !== simpleIcons[key].slug) {
    console.log(key)
  }
})
```

## SimpleIcons 7:

```js
import * as simpleIcons from 'simple-icons/icons'

Object.keys(simpleIcons).forEach(key => {
  if (key.slice(2).toLowerCase() !== simpleIcons[key].slug) {
    console.log(key)
  }
})
```

they both print out nothing, so I think we're just optimising for whatever makes the intent of the code the clearest.